Restructure of recommendations

msecret opened the pull request:
- Renamed "Recommendations".
- Restructure Recommendations section so it's paged similar to css.
- Added section on React.
- Telling people to never use bower.

Commit:
Restructure of recommendations

- Renamed "Recommendations".
- Restructure Recommendations section so it's paged similar to css.
- Added section on React.
- Telling people to never use bower.

mbland commented on pages/recommended/dependency_management.md:
```diff
@@ -0,0 +1,42 @@
+## Bower
+#### Do not use Bower.
+
+It's not needed and should be phased out and replaced by npm. More information can be found here: http://gofore.com/ohjelmistokehitys/stop-using-bower/
```
FWIW, our [accessibility guidelines](https://pages.18f.gov/accessibility/links/) suggest that the anchor text be more descriptive, such as `for more information, read the article [Why We Should Stop Using Bower – And How to Do It](http://gofore.com/ohjelmistokehitys/stop-using-bower/)`

Commit:
Update bower url to be a link

To make it more accessible and overall better.

Commit:
Move dependency management above mv frameworks

Commit:
Fix dependency management title

The dependency management page had “M**V Frameworks” as the title, so I
changed that and fixed the structure in `_config.yml` to ensure that
the current page state was displayed properly.

jeremiak commented:
@msecret Looks all good, just made a small correction. Going to merge myself since the content was good and no need to get another pair of :eyes:
